platform/security/sts/security-sts-ldapclaimshandler/src/main/java/ddf/security/sts/claimsHandler/ClaimsHandlerManager.java

```diff
@@ -193,6 +209,23 @@ public void update(Map<String, Object> props) {
 
   public void destroy() {}
 
+  protected ConnectionFactory createConnectionFactory(
+      List<String> urls, Boolean startTls, String loadBalancingAlgorithm) throws LdapException {
+    List<ConnectionFactory> connectionFactories = new ArrayList<>();
+
+    for (String singleUrl : urls) {
```

Comment:
✏️ This section could be simplified to use java streams. Also it looks like `connectionFactories` can be `final` too

```
final List<ConnectionFactory> connectionFactories = 
    urls
        .stream()
        .map(url -> createLdapConnectionFactory(new PropertyResolver(url).toString(), startTls);
        .collect(toList());
```